flyteadmin/pkg/errors/errors.go

```diff
@@ -109,8 +111,68 @@ func NewIncompatibleClusterError(ctx context.Context, errorMsg, curCluster strin
 	return statusErr
 }
 
-func NewWorkflowExistsDifferentStructureError(ctx context.Context, request *admin.WorkflowCreateRequest) FlyteAdminError {
-	errorMsg := "workflow with different structure already exists"
+func compareJsons(jsonArray1 jsondiff.Patch, jsonArray2 jsondiff.Patch) []string {
+	results := []string{}
+	map1 := make(map[string]jsondiff.Operation)
+	for _, obj := range jsonArray1 {
+		map1[obj.Path] = obj
+	}
+
+	for _, obj := range jsonArray2 {
+		if val, ok := map1[obj.Path]; ok {
+			result := fmt.Sprintf("%s: %s -> %s\t", obj.Path, obj.Value, val.Value)
+			results = append(results, result)
+		}
+	}
+	return results
+}
+
+func NewTaskExistsDifferentStructureError(ctx context.Context, request *admin.TaskCreateRequest, oldSpec *core.TaskTemplate, newSpec *core.TaskTemplate) FlyteAdminError {
+	errorMsg := "task with different structure already exists:\n"
+	// omit source code file object storage path
+	diff, _ := jsondiff.Compare(oldSpec, newSpec, jsondiff.Ignores("/Target/Container/args/2"))
```

Comment:
why do we need two Compare calls here?

it looks like according to https://pkg.go.dev/github.com/wI2L/jsondiff#Differ.Compare this should output the diff between src & tgt

Reply:
Because [jsondiff.Compare](https://pkg.go.dev/github.com/wI2L/jsondiff#Differ.Compare) returns a json patch file meaning a series of operations of that json.

It only show the modified `target` json other than `source` json. Since we're talking about `diff` message. It must show the difference between changes from `source` to `target`.

`diff = jsondiff.Compare(oldTemplate, newTemplate)`
`rdiff = jsondiff.Compare(newTemplate, oldTemplate`
which prefix `r` means `reverse`.

Finally `compareJsons()` zip the `diff` and its complements `rdiff`.